projects/noloan/app/src/main/java/hasadna/noloan/SpamRecyclerAdapter.java

```diff
@@ -29,26 +31,80 @@
+      for (SmsMessage sms : SmsMessages.get().getDbMessages()) {
+        if (SmsMessages.get().getInboxMessages().get(i).getBody().contentEquals(sms.getBody())
+            && SmsMessages.get()
+                .getInboxMessages()
+                .get(i)
+                .getSender()
+                .contentEquals(sms.getSender())) {
+          // Add message from the DB - keep received date of user's message
+          inboxSpam.add(
+              sms.toBuilder()
+                  .setReceivedAt(SmsMessages.get().getInboxMessages().get(i).getReceivedAt())
+                  .build());
+        }
+      }
+    }
 
-    // Listen to db smsMessages
+    // Listen to db smsMessages. Update just relevant messages that are in the inbox as well.
+    // Keep the received date of the user's message.
     SmsMessages.get()
         .setMessagesListener(
             new SmsMessages.MessagesListener() {
               @Override
               public void messageAdded(SmsMessage newMessage) {
-                handler.post(() -> notifyItemInserted(SmsMessages.get().getDbMessages().size()));
+                int inboxIndex =
+                    SmsMessages.get()
+                        .searchMessage(newMessage, SmsMessages.get().getInboxMessages());
+                if (inboxIndex != -1) {
+                  inboxSpam.add(
+                      newMessage
+                          .toBuilder()
+                          .setReceivedAt(
+                              SmsMessages.get().getInboxMessages().get(inboxIndex).getReceivedAt())
+                          .build());
+                  handler.post(() -> notifyItemInserted(inboxSpam.size()));
+                }
               }
 
               @Override
               public void messageModified(int index) {
-                handler.post(() -> notifyItemChanged(index));
+                int inboxSpamIndex =
+                    SmsMessages.get()
+                        .searchMessage(SmsMessages.get().getDbMessages().get(index), inboxSpam);
+                if (inboxSpamIndex != -1) {
+                  inboxSpam.set(
+                      inboxSpamIndex,
+                      SmsMessages.get()
+                          .getDbMessages()
+                          .get(index)
+                          .toBuilder()
+                          .setReceivedAt(
+                              SmsMessages.get()
+                                  .getInboxMessages()
+                                  .get(inboxSpamIndex)
+                                  .getReceivedAt())
+                          .build());
+                  handler.post(() -> notifyItemChanged(inboxSpamIndex));
+                }
               }
 
               @Override
               public void messageRemoved(int index, SmsMessage smsMessage) {
-                handler.post(() -> notifyItemRemoved(index));
+                int inboxSpamIndex = SmsMessages.get().searchMessage(smsMessage, inboxSpam);
```

Comment:
It can be call index, since there is not other indexes around.
Same in modified.

Reply:
`index` is part of the function's parameter. 

`inboxSpamIndex` feels easier and faster to "grasp" to which list this index belongs to IMO.

Reply:
> index is part of the function's parameter.

Right...
I prefer shorter names so it bother me.